packages/next-app/src/app/daoadmin/action/[actionId]/(components)/ActionAttributeContent.tsx

```diff
@@ -1,37 +1,83 @@
+import { useNetwork } from "wagmi";
 
 import { Button } from "#/components";
-import { AdminToolsContext, toSlug } from "#/contexts/AdminToolsContext";
+import { toCammelCase, useAdminTools } from "#/contexts/AdminToolsContext";
+import { fetchExistingPool } from "#/utils/fetcher";
 import { truncateAddress } from "#/utils/truncateAddress";
 
+const Input = React.forwardRef<
+  HTMLInputElement,
+  React.HTMLProps<HTMLInputElement>
+>(({ label, ...rest }: React.HTMLProps<HTMLInputElement>, ref) => {
+  return (
+    <div>
+      <label className="mb-2 block text-sm text-gray-400">{label}</label>
+      <input
+        ref={ref}
+        {...rest}
+        className="selection:color-white box-border inline-flex h-[35px] w-full appearance-none items-center justify-center rounded-[4px] bg-blackA5 px-[10px] text-[15px] leading-none text-white shadow-[0_0_0_1px] shadow-blackA9 outline-none selection:bg-blackA9 hover:shadow-[0_0_0_1px_black] focus:shadow-[0_0_0_2px_black] disabled:bg-blackA9"
+      />
+    </div>
+  );
+});
+
+const CheckPoolID = ({
+  fieldName,
+  poolName,
+  poolError,
+}: {
+  fieldName: string;
+  poolName: string | undefined;
+  poolError: string | undefined;
+}) => {
+  return (
+    <div className="mt-2 flex gap-1 text-sm text-gray-400">
+      {fieldName === "Pool ID" && poolName ? (
+        <>
+          <h1 className="text-white">Pool Name:</h1>
+          <h1>{poolName}</h1>
+        </>
+      ) : fieldName === "Pool ID" && poolError ? (
+        <h1>{poolError}</h1>
+      ) : (
+        <></>
+      )}
+    </div>
+  );
+};
+
 export function ActionAttributeContent() {
-  const { register, handleSubmit } = useForm();
+  const { register, handleSubmit, watch } = useForm();
   // eslint-disable-next-line no-console
   const onSubmit = (data: unknown) => console.log(data);
   const { push } = useRouter();
-  const { selectedAction } = useContext(AdminToolsContext);
+  const { selectedAction } = useAdminTools();
+  const { chain } = useNetwork();
+  const [poolName, setPoolName] = React.useState<string>();
+  const [poolError, setPoolError] = React.useState<string>();
 
-  const Input = React.forwardRef<HTMLInputElement, HTMLProps<HTMLInputElement>>(
-    ({ label, ...rest }: React.HTMLProps<HTMLInputElement>, ref) => {
-      return (
-        <div>
-          <label className="mb-2 block text-sm font-bold text-gray-400">
-            {label}
-          </label>
-          <input
-            ref={ref}
-            {...rest}
-            className="selection:color-white box-border inline-flex h-[35px] w-full appearance-none items-center justify-center rounded-[4px] bg-blackA5 px-[10px] text-[15px] leading-none text-white shadow-[0_0_0_1px] shadow-blackA9 outline-none selection:bg-blackA9 hover:shadow-[0_0_0_1px_black] focus:shadow-[0_0_0_2px_black] disabled:bg-blackA9"
-          />
-        </div>
-      );
+  const poolID = watch("poolID");
+
+  React.useEffect(() => {
+    if (poolID) {
```

Comment:
in this case you can return early instead of doing this do avoid more indents to your code and requiring the reader to keep the whole control flow context in mind:

```suggestion
    if (!ethers.utils.isAddress(poolId)) return;
```